### PR TITLE
Add docs for Device Trust tctl commands

### PR DESCRIPTION
**Pull request by xacrimon**
This PR adds some docs to document the support for the `device` resource implemented by https://github.com/gravitational/teleport/pull/21921.

Updates https://github.com/gravitational/teleport.e/issues/829 thus also https://github.com/gravitational/teleport.e/issues/514.

### CODE REVIEW

File: docs/pages/includes/device-spec.mdx [hunk 1]
```diff
@@ -0,0 +1,35 @@
+```
```
Comment (ptgott) [inline docs/pages/includes/device-spec.mdx:1]: Should this be a YAML file? That way, we can also remove the opening/closing triple-backticks.

File: docs/pages/reference/resources.mdx [hunk 2]
```diff
@@ -196,3 +203,9 @@ Roles govern access to databases, SSH servers, Kubernetes clusters, web services
 Login rules contain logic to transform SSO user traits during login.
 
 (!docs/pages/includes/login-rule-spec.mdx!)
+
+### Device
+
+Device contains information identifying a trusted device.
+
+(!docs/pages/includes/device-spec.mdx!)
```
Comment (ptgott) [inline docs/pages/reference/resources.mdx:211]:
````suggestion

```yaml
(!docs/pages/includes/device-spec.yaml!)
```
````
Assuming we change this to a YAML file

File: docs/pages/includes/device-spec.mdx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  # os_type is the operating system type of the device.
+  # This must be one of "linux", "windows", or "macos".
+  os_type: macos
+
+  # credential is the device credential used by the device to authenticate with Teleport.
+  credential:
+    # id is the unique identifier of the credential.
+    id: 000000-0000-0000-0000-000000000000
+
+    # public_key_der is the public part of the asymmetric keypair used to verify authenticity.
```
Comment (zmb3) [inline docs/pages/includes/device-spec.mdx:29]: Worth mentioning that it's base64-encoded here? (DER is a binary format)

**Review by zmb3 — COMMENTED** [review]
We should also update the CLI reference page.

File: docs/pages/reference/resources.mdx [hunk 4]
```diff
@@ -141,6 +142,12 @@ $ tctl rm github/myteam
 
 # Delete a local user called "admin":
 $ tctl rm users/admin
+
+# Show all devices:
+$ tctl get devices
+
+# Fetch a specific device:
+$ tctl get devices/<asset-tag>
```
[the remark this answers is not in the document]
Comment (xacrimon) [reply docs/pages/reference/resources.mdx:150]: Both work as per the Device Trust RFD. This is provided as an example but generally
1. if an ID match is present, return that
2. attempt to search for a matching asset tag, return multiple if there are matching one

File: docs/pages/includes/device-spec.mdx (hunk 1 above)
Comment (xacrimon) [reply docs/pages/includes/device-spec.mdx:1]: From what I see, all other includes here are `.mdx` files containing a single codeblock so I followed existing convention. Should we do anything differently and switch them all out?

File: docs/pages/includes/device-spec.mdx [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+kind: device
+version: v1
+metadata:
+  # name is a unique random identifier for the device in the cluster.
+  name: 5ff09619-527c-4a17-973f-cd9cd5c93990
+spec:
+  # asset_tag is a user controlled identifier for the device. This is usually a device serial number.
+  asset_tag: HD6M74XNCK
+
+  # collected_data is a list of data recently collected from the device.
```
[the remark this answers is not in the document]
Comment (xacrimon) [reply docs/pages/includes/device-spec.mdx:11]: It is a YAML list. I've added an example.

**Comment by xacrimon** [comment]
> We should also update the CLI reference page.

I added the `device` resource to some of the "for example" insertions in the CLI reference.

File: docs/pages/includes/device-spec.mdx [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+  # Each list item is an object containing event and device metadata..
+  collected_data:
+    # collect_time is the device-recorded timestamp of when the data was collected.
+  - collect_time:
+      nanos: 312862000
+      seconds: 1677099866
```
Comment (codingllama) [inline docs/pages/includes/device-spec.mdx:17]: Do you think we should convert this timestamp too, as we do the others? (Follow up for another PR, most likely.)

File: docs/pages/includes/device-spec.mdx [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+  update_time: "2023-02-22T21:04:26.396471Z"
+
+  # os_type is the operating system type of the device.
+  # This must be one of "linux", "windows", or "macos".
+  os_type: macos
+
+  # credential is the device credential used by the device to authenticate with Teleport.
+  credential:
+    # id is the unique identifier of the credential.
+    id: 000000-0000-0000-0000-000000000000
```
Comment (codingllama) [inline docs/pages/includes/device-spec.mdx:43]:
```suggestion
    id: ebe41e2b-5dbb-40a0-82fb-cded7d3373dd
```

Just so it looks more "normal".

**Review by codingllama — APPROVED** [review]
Thanks for the update, Joel!

File: docs/pages/reference/resources.mdx (hunk 4 above)
[the remark this answers is not in the document]
Comment (zmb3) [reply docs/pages/reference/resources.mdx:150]: Awesome!

**Review by zmb3 — APPROVED** [review]
Thanks Joel!

File: docs/pages/includes/device-spec.mdx (hunk 1 above)
Comment (ptgott) [reply docs/pages/includes/device-spec.mdx]: Some docs pages include YAML files within `yaml` code snippets:

```
$ grep -RnE '\(![^!]+\.yaml!)' docs/pages | wc -l
28
```

It's not strictly necessary if the page renders correctly, but I do think it's a cleaner/more explicit approach if the partial consists only of a YAML document.

**Comment by xacrimon** [comment]
@zmb3 Give me a moment on this one. I will update the timestmap format as suggested by Alan after a discussion about how we should handle timestamp conversions and other serialization differences without gogo going forward.

**Comment by xacrimon** [comment]
@codingllama I've now opened https://github.com/gravitational/teleport/pull/22198 and updated the example data to get nicely formatted timestamps.

**Review by codingllama — APPROVED** [review]

**Pull request merged**